src/utils/search-bar.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Autocomplete, TextField } from '@mui/material';
+import { fetchDirectoryContent, searchElementsInfos } from './rest-api';
+import {
+    getFileIcon,
+    useDebounce,
+    useSnackMessage,
+} from '@gridsuite/commons-ui';
+import { Search } from '@mui/icons-material';
+import { useDispatch, useSelector } from 'react-redux';
+import { setSelectedDirectory, setTreeData } from '../redux/actions';
+import Grid from '@mui/material/Grid';
+import { updatedTree } from '../components/tree-views-container';
+import Typography from '@mui/material/Typography';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+const styles = {
+    icon: (theme) => ({
+        marginRight: theme.spacing(2),
+        width: '18px',
+        height: '18px',
+    }),
+    grid: {
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        display: 'inline-block',
+    },
+    grid2: (theme) => ({
+        marginRight: theme.spacing(2),
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        display: 'inline-block',
+        color: 'grey',
+    }),
+};
+export const SEARCH_FETCH_TIMEOUT_MILLIS = 1000; // 1 second
+
+export const SearchBar = ({ inputRef }) => {
+    const dispatch = useDispatch();
+    const { snackError } = useSnackMessage();
+    const [elementsFound, setElementsFound] = useState([]);
+    const lastSearchTermRef = useRef('');
+    const [loading, setLoading] = useState(false);
+    const treeData = useSelector((state) => state.treeData);
+    const treeDataRef = useRef();
+    const intl = useIntl();
+    treeDataRef.current = treeData;
+    const searchMatchingEquipments = useCallback(
+        (searchTerm) => {
+            lastSearchTermRef.current = searchTerm;
+            searchTerm &&
+                searchElementsInfos(searchTerm)
+                    .then((infos) => {
+                        if (infos.length) {
+                            setElementsFound(infos);
+                        } else {
+                            setElementsFound([]);
+                        }
+                    })
+                    .catch((error) => {
+                        snackError({
+                            messageTxt: error.message,
+                            headerId: 'elementsSearchingError',
+                        });
+                    });
+        },
+        [snackError]
+    );
+
+    const debouncedSearchMatchingElements = useDebounce(
+        searchMatchingEquipments,
+        SEARCH_FETCH_TIMEOUT_MILLIS
+    );
+
+    const handleChangeInput = (newId) => {
```

Comment:
using useCallback for rendering jsx is not typically recommended.
if you want to prevent unecessary re-rendres of a component you should consider using React.memo and extracting the content of this function into a small component ( you can add it in the buttom of this file)